File: ParkingDb/Models/ParkingArea.cs
using ParkingDb.Interfaces;

namespace ParkingDb.Models;

public class ParkingArea : DbObject<string>, IActive
{
    public string Name { get; set; } = string.Empty;
    public string Location { get; set; } = string.Empty;
    public decimal WeekdaysHourlyRateUsd { get; set; } = 0.0m;
    public decimal WeekendHourlyRateUsd { get; set; } = 0.0m;
    public double DiscountPercentage { get; set; } = 0.0;
    public string? Description { get; set; }
    public bool IsActive { get; set; }
}

File: parkingflow.client/src/models/ParkingArea.ts
export interface ParkingArea {
    id?: string;
    name: string;
    location: string;
    weekdaysHourlyRateUsd: number;
    weekendHourlyRateUsd: number;
    discountPercentage: number;
    description: string | null;
    isActive: boolean;
}